privacyidea/static/components/token/controllers/tokenDetailController.js

```diff
@@ -190,9 +190,42 @@ myApp.controller("tokenDetailController", function ($scope,
         }, $scope.get);
     };
 
-    $scope.delete = function () {
-        TokenFactory.delete($scope.tokenSerial, $scope.return_to);
+    $scope.deleteTokenAsk = function() {
+            $('#dialogTokenDelete').modal();
+        var tokenType = $scope.token.info.tokenkind;
+        if (tokenType == "hardware"){
+        $('#dialogTokenDelete').modal();
```

Comment:
This looks good, but please fix the indentation. It is really hard to read otherwise. the code should look something like this:
```
if (condition) {
    doSomething();
} else {
    doSomethingElse();
}
```